openapi/paths.go

```diff
@@ -206,6 +269,13 @@ func (p *PathItem) Validate(ctx context.Context, opts ...validation.Option) []er
 	core := p.GetCore()
 	errs := []error{}
 
+	o := validation.NewOptions(opts...)
+
+	openapi := validation.GetContextObject[OpenAPI](o)
```

Comment:
I was actually thinking we could maybe be a bit less strict about needing the OpenAPI object to be passed in here. Instead of panic we can just assume when we don't receive the OpenAPI we are working with a document of the latest version.

Was thinking through our future use cases for a linter and more robust validation in the future and we will prob want to double down on support people being able to call the `Validate` methods on each object manually even when they don't have a full document object yet.